src/libsync/propagateupload.h

```diff
@@ -220,5 +259,38 @@ private slots:
     void abortWithError(SyncFileItem::Status status, const QString &error);
 };
 
+/**
+ * @brief The PropagateBundleQNAM class
+ * @ingroup libsync
+ */
+class PropagateBundleQNAM : public PropagateItemJob {
+    Q_OBJECT
+
+private:
+    QVector<AbstractNetworkJob*> _jobs; /// network jobs that are currently in transit
+    QVector<SyncFileItemPtr> _bundledFiles;
+    QJsonObject _bundledFilesMetadata;
+    quint64 _contentID = 1;
+    quint64 chunkSize() const { return _propagator->chunkSize(); }
+
+    bool _finished; // Tells that all the jobs have been finished
+
+public:
+    PropagateBundleQNAM(OwncloudPropagator* propagator)
+        : PropagateItemJob(propagator, SyncFileItemPtr(new SyncFileItem)) {}
+    void start() Q_DECL_OVERRIDE;
+    bool append(SyncFileItemPtr bundledFile);
```

Comment:
object that hae a copy constructor should preferably be taken as const reference.
So `const SyncFileItemPtr &bundledFile`